frontend/packages/console-app/src/components/volume-group-snapshot/create-volume-group-snapshot.tsx

```diff
@@ -0,0 +1,328 @@
+  const [groupSnapshotClassName, setGroupSnapshotClassName] = React.useState('');
+  const title = t('console-app~Create VolumeGroupSnapshot');
+  const [labelExpressions, setLabelExpressions] = React.useState<MatchExpression[]>([]);
+  const [labels, setLabels] = React.useState<{ [key: string]: string[] }>({});
+  const [validationError, setValidationError] = React.useState<string | null>(null);
+
+  const resourceWatch = React.useMemo(() => {
+    const watch = {
+      kind: referenceForModel(PersistentVolumeClaimModel),
+      isList: true,
+      namespace,
+      selector: labelExpressions.length > 0 ? { matchExpressions: labelExpressions } : undefined,
+    };
+
+    return watch;
+  }, [namespace, labelExpressions]);
+
+  const [data, loaded, loadError] = useK8sWatchResource<PersistentVolumeClaimKind[]>(resourceWatch);
+
+  const [storageClasses, setStorageClasses] = React.useState<StorageClassResourceKind[]>([]);
+  const storageClassWatch = {
+    kind: StorageClassModel.kind,
+    isList: true,
+  };
+
+  const [storageClassData] = useK8sWatchResource<StorageClassResourceKind[]>(storageClassWatch);
+
+  React.useEffect(() => {
+    if (storageClassData) {
+      setStorageClasses(storageClassData);
+    }
+  }, [storageClassData]);
+
+  const getProvisioner = React.useCallback(
+    (storageClassName: string): string | null => {
+      if (!storageClassName || !storageClasses.length) return null;
+
+      const storageClass = storageClasses.find((sc) => sc.metadata.name === storageClassName);
+      return storageClass?.provisioner || null;
+    },
+    [storageClasses],
+  );
+
+  const validatePVCs = React.useCallback(
+    (pvcs: PersistentVolumeClaimKind[]) => {
+      if (!pvcs || pvcs.length <= 1 || labelExpressions.length === 0) {
+        setValidationError(null);
+        return;
+      }
+
+      const provisioners = new Set();
+      pvcs.forEach((pvc) => {
+        const storageClassName = pvc.spec?.storageClassName;
+        if (storageClassName) {
+          const provisioner = getProvisioner(storageClassName);
+          if (provisioner) {
+            provisioners.add(provisioner);
+          }
+        }
+      });
+
+      if (provisioners.size > 1) {
+        setValidationError(
+          t(
+            'console-app~Selected PVCs have different storage provisioners. All PVCs must have StorageClasses with the same provisioner.',
+          ),
+        );
+        return;
+      }
+
+      setValidationError(null);
+    },
+    [getProvisioner, labelExpressions.length, t],
+  );
+
+  React.useEffect(() => {
+    if (loaded && !loadError) {
+      const newPvcObjs = data || [];
+      setPVCObjs(newPvcObjs);
+
+      // Validate PVCs for same provisioner
+      if (newPvcObjs.length > 1) {
+        validatePVCs(newPvcObjs);
+      } else {
+        setValidationError(null);
+      }
+    }
+  }, [data, loaded, loadError, validatePVCs]);
```

Comment:
same here... no need of extra `useState` and extra re-render...
```suggestion
const validationError =  React.useMemo(() => {
    if (loaded && !loadError) {
      const newPvcObjs = data || [];
      setPVCObjs(newPvcObjs);

      // Validate PVCs for same provisioner
      if (newPvcObjs.length > 1) {
        return validatePVCs(newPvcObjs);
      } else {
        return null;
      }
    }
  }, [data, loaded, loadError, validatePVCs]);
```